recipes/configs/llama2/1B_full_ppo.yaml

```diff
@@ -0,0 +1,182 @@
+  _component_: torchtune.datasets.text_completion_dataset
+  source: trl-internal-testing/sentiment-trl-style
+  max_seq_len: null
+  split: train
+  column: prompt
+  add_eos: False
+
+
+# manually constructing 1B models
+policy_model:
+  _component_: torchtune.models.llama2.llama2
+  vocab_size: 32000
+  num_layers: 22
+  num_heads: 32
+  num_kv_heads: 4
+  embed_dim: 2048
+  max_seq_len: 2048
+  intermediate_dim: 5632
+  attn_dropout: 0.0
+  norm_eps: 1e-5
+
+reward_and_value_model:
+  _component_: torchtune.models.llama2.llama2_classifier
+  num_classes: 1
+  vocab_size: 32000
+  num_layers: 22
+  num_heads: 32
+  num_kv_heads: 4
+  embed_dim: 2048
+  max_seq_len: 2048
+  intermediate_dim: 5632
+  attn_dropout: 0.0
+  norm_eps: 1e-5
+
+# checkpointer for the policy model - update this if resuming from checkpoint
+checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir: /tmp/TinyLlama_v1.1
+  checkpoint_files: [
+      "pytorch_model.bin",
+  ]
+  # this is the only place where you should update `recipe_checkpoint` if resuming training
+  recipe_checkpoint: null
+  output_dir: ${output_dir}/policy
+  model_type: LLAMA2
+
+# this should be setup identically to the policy model checkpointer at the start of training
+# ensure `checkpoint_files` always points to the original policy weights, even if resuming training
+ref_policy_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir: /tmp/TinyLlama_v1.1
+  checkpoint_files: [
+      "pytorch_model.bin",
+  ]
+  output_dir: ${output_dir}/policy
+  model_type: LLAMA2
+
+# checkpointer for the value model - update this if resuming from checkpoint
+# since this model will be identical to the reward model it's helpful to initialise this
+# from the trained reward model weights
+value_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir:  /tmp/tinyllama_rm_sentiment_1b
+  # only `checkpoint_files` need to be updated if resuming training
+  checkpoint_files: [
+      "model.safetensors"
+  ]
+  output_dir: ${output_dir}/value
+  model_type: MISTRAL_REWARD
+
+# checkpointer for the reward model, ensure `checkpoint_files`
+# always points to the original reward model weights, even if resuming training
+reward_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir:  /tmp/tinyllama_rm_sentiment_1b
+  checkpoint_files: [
+      "model.safetensors"
+  ]
+  output_dir: ${output_dir}/value
+  model_type: MISTRAL_REWARD
+
+
+# Training env
+device: cuda
+
+# Training arguments
+batch_size: 256
+num_steps: 100000
+ppo_epochs: 2
+ppo_batch_size: 128
+gradient_accumulation_steps: 8
+
+# Memory management and performance
+compile: True
+optimizer:
+  _component_: torch.optim.AdamW
+  weight_decay: 0.01
+  lr: 3e-6
+
+optimizer_in_bwd: False
+log_peak_memory_stats: False
+enable_activation_checkpointing: True
+
+# Reduced precision
+dtype: bf16
+
+# Trajectory generation arguments
```

Comment:
What's this?

Reply:
Sorry, could be clear, should be a high-level header comment for the next block of args